dfdaemon/proxy/proxy.go

```diff
@@ -66,15 +77,28 @@ func New(opts ...Option) (*TransparentProxy, error) {
 	return proxy, nil
 }
 
+// NewFromConfig returns a new transparent proxy from the given properties
+func NewFromConfig(c config.Properties) (*TransparentProxy, error) {
+	opts := []Option{
+		WithRules(c.Proxies),
+		WithHTTPSHosts(c.HijackHTTPS.Hosts...),
```

Comment:
I'll add a check before using `HijackHTTPS`. Using struct instead of pointer is also a valid choice, but I think using pointer will be easier to tell whether user has configured the field.